project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/gradle/GradleCompilationPropertyChecker.kt

```diff
@@ -0,0 +1,86 @@
+package org.plan.research.minimization.plugin.execution.gradle
+
+import arrow.core.Either
+import arrow.core.raise.either
+import arrow.core.raise.ensure
+import arrow.core.raise.ensureNotNull
+import com.intellij.execution.ExecutionTargetManager
+import com.intellij.execution.RunManager
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.execution.runners.ExecutionEnvironmentBuilder
+import com.intellij.execution.runners.ProgramRunner
+import com.intellij.openapi.application.writeAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.guessProjectDir
+import kotlinx.coroutines.CoroutineScope
+import org.gradle.tooling.model.GradleTask
+import org.jetbrains.plugins.gradle.service.execution.GradleExecutionHelper
+import org.jetbrains.plugins.gradle.service.execution.GradleExternalTaskConfigurationType
+import org.jetbrains.plugins.gradle.service.execution.GradleRunConfiguration
+import org.jetbrains.plugins.gradle.util.GradleConstants
+import org.plan.research.minimization.plugin.errors.CompilationPropertyCheckerError
+import org.plan.research.minimization.plugin.model.CompilationPropertyChecker
+
+class GradleCompilationPropertyChecker(private val cs: CoroutineScope) : CompilationPropertyChecker {
+    override suspend fun checkCompilation(project: Project) = either {
```

Comment:
Let's refactor the return type a bit. Returning Throwable is quite confusing and not scalable, so let's introduce an interface or abstract class of compilation errors and put it as a return type (without generics). Each PropertyChecker will provide its own compilation errors, which can be compared to each other specifically (e.g. overrides equals).